addons/sale_margin/models/sale_order.py

```diff
@@ -86,3 +86,9 @@ def _compute_margin(self):
             for order in self:
                 order.margin = mapped_data.get(order.id, 0.0)
                 order.margin_percent = order.amount_untaxed and order.margin/order.amount_untaxed
+
+    @api.returns('self', lambda value: value.id)
+    def copy(self, default=None):
+        new_order = super().copy(default=default)
```

Comment:
Try always to document the why of the things.

```suggestion
    def copy(self, default=None):
        """Launch recomputation of the purchase price to get
        an updated result for the new sales order.
        """    
        new_order = super().copy(default=default)
```

Reply:
Oh, I added my own comment before reading yours :sweat_smile: